File: java/src/test/java/challenges/aoc2024/day5/model/ProductionUpdateTest.java
package challenges.aoc2024.day5.model;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.List;

import org.junit.jupiter.api.Test;

class ProductionUpdateTest {

    @Test
    void shouldBuildProductionUpdateObjectFromString() {
        // When
        final ProductionUpdate productionUpdate = ProductionUpdate.of("1,23,456");

        // Then
        assertEquals(ImmutableProductionUpdate.class, productionUpdate.getClass());

        final List<Integer> pages = productionUpdate.getPages();
        assertEquals(3, pages.size());
        assertEquals(1, pages.get(0));
        assertEquals(23, pages.get(1));
        assertEquals(456, pages.get(2));
    }
}

File: scala/src/test/scala/challenges/aoc2024/day5/model/ProductionUpdateTest.scala
package challenges.aoc2024.day5.model

import org.scalatest.funsuite.AnyFunSuite

class ProductionUpdateTest extends AnyFunSuite {

  test("should build ProductionUpdate object from string") {
    // When
    val productionUpdate: ProductionUpdate = ProductionUpdate.of("1,23,456")

    // Then
    assert(productionUpdate.isInstanceOf[ProductionUpdate])

    val pages: List[Int] = productionUpdate.pages
    assert(3 == pages.size)
    assert(1 == pages.head)
    assert(23 == pages(1))
    assert(456 == pages(2))
  }
}
